integration/tests/cook/test_multi_user.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import pytest
+import unittest
+
+from tests.cook import util
+
+@unittest.skipUnless(util.multi_user_tests_enabled(), "Requires using multi-user coniguration (e.g., BasicAuth) for Cook Scheduler")
+@pytest.mark.timeout(util.DEFAULT_TEST_TIMEOUT_SECS)  # individual test timeout
+class MultiUserCookTest(unittest.TestCase):
+    _multiprocess_can_split_ = True
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cook_url = util.retrieve_cook_url()
+        util.init_cook_session(cls.cook_url)
+
+    def setUp(self):
+        self.cook_url = type(self).cook_url
+        self.logger = logging.getLogger(__name__)
+        self.user_factory = util.UserFactory(self)
+
+    def test_job_delete_permission(self):
+        user1, user2 = self.user_factory.new_users(2)
+        with user1:
+            job_uuid, resp = util.submit_job(self.cook_url, command='sleep 10')
+        try:
+            self.logger.debug(f"User 1: {user1.name}")
+            self.assertEqual(resp.status_code, 201, resp.text)
+            with user2:
+                util.kill_jobs(self.cook_url, [job_uuid], expected_status_code=403)
+            with user1:
+                util.kill_jobs(self.cook_url, [job_uuid])
```

Comment:
Should we assert after this that the job was killed?

Reply:
Done